### PR TITLE
Add salary field

### LINKED ISSUE #1320: Add salary field to satisfy Google's Job Search schema
Assuming a user fills out all the job form fields correctly, Google will still throw a warning (_not_ an error) because listings do not have a salary by default. We should consider adding this:

https://wpjobmanager.com/document/tutorial-adding-a-salary-field-for-jobs/

... to the core plugin, IMHO.

We should also consider renaming the `Location` field to `Address`, since if you don't enter an exact street address, Google will throw other warnings (or actual errors, depending on how the user enters the location). `Address` is less ambiguous in this regard.

Related to #1313

### PR DESCRIPTION
**Pull request by onubrooks**
Fixes #1320 
Builds upon #1510 

### Changes proposed in this Pull Request

* Adds a salary field to the frontend and admin forms.
* Adds salary information to Google's Job Search Structured Data
* Adds a setting to enable salary field. This is useful to not break sites using snippets or other plugins to add salary field.

### Testing instructions

* Create/Edit Job Listing and add salary
* View the salary on the frontend
* Copy the output HTML and paste here - [https://search.google.com/structured-data/testing-tool/u/0/](https://search.google.com/structured-data/testing-tool/u/0/)

### New/Updated Hooks

* `wpjm_job_salary_currency`: This filter provides a default currency, `USD`.
* `wpjm_job_salary_unit`: This filter provides a default unit, `YEAR`, which can be adjusted to `MONTH`, `WEEK` etc.
* `the_job_salary`: Filters the returned job salary.

### Improvements
Base on https://github.com/Automattic/WP-Job-Manager/issues/1320#issuecomment-537865339, here are a few possible enhancements:

- Setting for per job currencies with a global default. Base our function(s) off of get_woocommerce_currencies and get_woocommerce_currency_symbol.
- Interval will also need to be supported / tracked (HOUR, DAY, WEEK, MONTH, YEAR). For the default, we could probably use YEAR with a filter to adjust.
- Ideally, ranges should also be supported from the start. We could do some magic parsing of a string (20000-40000) -> 'minValue' => 20000, 'maxValue' => 40000 or two fields. Two fields would probably be best.
- UI: input currency, salary value/range (seamless way to go between a single salary and range), Interval unit (hour, day, week, month, year)

### CODE REVIEW

File: includes/class-wp-job-manager-post-types.php [hunk 1]
```diff
@@ -702,6 +703,9 @@ public function job_feed_item() {
 		if ( $company ) {
 			echo '<job_listing:company><![CDATA[' . esc_html( $company ) . "]]></job_listing:company>\n";
 		}
+		if ( $salary ) {
+			echo '<job_listing:salary><![CDATA[' . esc_html( $salary ) . "]]></job_listing:salary>\n";
+		}
```
Comment (dadish) [inline includes/class-wp-job-manager-post-types.php:708]: Couldn't understand what's this used for.

**Review by dadish — APPROVED** [review]
Works great! 🔥

File: includes/class-wp-job-manager-post-types.php (hunk 1 above)
Comment (onubrooks) [reply includes/class-wp-job-manager-post-types.php:708]: Not sure either but I think it has something to do with RSS feeds

**Comment by tripflex** [comment]
https://wordpress.org/support/topic/new-update-adds-salary-field-but-how-to-get-data-from-old-salary-field/

As i'm sure others will point out, this probably was not the best idea to output the rate and currency as some people use this field to define the salary in other means, and there's no easy way of disabling or removing this.

**Comment by StefMattana** [comment]
There seems to be a conflict with a third-party plugin too (WPJM Extra Fields), as the core salary filed keeps being displayed despite the option being toggled off: https://wordpress.org/support/topic/conflict-with-alternate-plugin-for-showing-salary/#post-15764240

I asked the user to report this to the plugin author, but I believe that's worth flagging it here too.

**Comment by StefMattana** [comment]
Upon testing 1.36.2 I found out that nothing gets displayed in my test site's job listings upon enabling the salary options. 

I filled the salary field both from the backend and the frontend, and also by adding simple numbers (es 1000) and numbers with specifications (1000/hours; 1000/year). Still, nothing got published. 

I am running JobifyVersion: 3.11.1 if that helps.

**Comment by StefMattana** [comment]
We already have a request [from the forum](https://wordpress.org/support/topic/all-salaries-are-now-showing-usd-year-after-salary-how-to-remove/#post-15763297) for implementing the `yearly/monthly/daily/hourly` toggle upon adding salaries via this new feature, which ideally could be customisable for each job entry

>we have some hourly, daily or monthly pay rates and these will vary from job to job, rather than being consistent across our platform for all jobs.

**Pull request merged**